frontend/src/lib/http/index.js

```diff
@@ -0,0 +1,25 @@
+export class NetworkError extends Error {
```

Comment:
Note to self: a bad name, I typically call this `FetchError`, don't know why I deviated